tests/csapi/txnid_test.go

```diff
@@ -0,0 +1,178 @@
+package csapi_tests
+
+import (
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/runtime"
+	"github.com/tidwall/gjson"
+)
+
+func mustHaveTransactionID(t *testing.T, roomID, eventID string) client.SyncCheckOpt {
+	return client.SyncTimelineHas(roomID, func(r gjson.Result) bool {
+		if r.Get("event_id").Str == eventID {
+			if !r.Get("unsigned.transaction_id").Exists() {
+				t.Fatalf("Event %s in room %s should have a 'transaction_id', but it did not", eventID, roomID)
+			}
+
+			return true
```

Comment:
You don't ever seem to assert that the `transaction_id` is actually the thing you gave originally? The type isn't even asserted currently so this would pass:
```js
{
  content: { ... },
  unsigned: {
    transaction_id: {
      key: "val"
    }
  }
}
```

Reply:
This was intentional as the "local echo" the presence is the pertinent aspect. However, I will add this for completeness.